services-js/commissions-search/src/App.js

```diff
@@ -30,33 +31,63 @@ class App extends React.Component {
     const target = event.target;
     const checked = target.checked;
     const name = target.name; //area id
+    const nameval = target.getAttribute('nameval'); //input label name
 
     let currentAreas;
+    let checkedNames;
 
     if (checked === false || checked === null) {
       // remove unchecked key
       currentAreas = update(this.state.currentAreas, { $unset: [name] });
+      //remove unchecked name
+      checkedNames = update(this.state.checkedNames, { $unset: [name] });
     } else {
       // add checked key
       currentAreas = update(this.state.currentAreas, {
         [name]: { $set: checked }, // value checked
       });
+      // add checked key name
+      checkedNames = update(this.state.checkedNames, {
+        [name]: { $set: nameval }, // value checked
+      });
     }
 
     this.setState({ currentAreas });
+    this.setState({ checkedNames });
   };
 
   handleOptionChange = event => {
     this.setState({ currentSeats: event.target.value });
   };
 
+  //get checkbox names from state and send to GA
+  findCheckedBoxes4GA = event => {
+    const chkArr = { ...this.state.checkedNames };
```

Comment:
You don't need to splat this into a new Object.

Also, `Arr` is not a good name for this since it's an object, not an array.